dataRetriever/resolvers/validatorInfoResolver.go

```diff
@@ -0,0 +1,188 @@
+	Marshalizer          marshal.Marshalizer
+	AntifloodHandler     dataRetriever.P2PAntifloodHandler
+	Throttler            dataRetriever.ResolverThrottler
+	ValidatorInfoPool    storage.Cacher
+	ValidatorInfoStorage storage.Storer
+	IsFullHistoryNode    bool
+}
+
+// validatorInfoResolver is a wrapper over Resolver that is specialized in resolving validator info requests
+type validatorInfoResolver struct {
+	dataRetriever.TopicResolverSender
+	messageProcessor
+	baseStorageResolver
+	validatorInfoPool    storage.Cacher
+	validatorInfoStorage storage.Storer
+}
+
+// NewValidatorInfoResolver creates a validator info resolver
+func NewValidatorInfoResolver(args ArgValidatorInfoResolver) (*validatorInfoResolver, error) {
+	err := checkArgs(args)
+	if err != nil {
+		return nil, err
+	}
+
+	return &validatorInfoResolver{
+		TopicResolverSender: args.SenderResolver,
+		messageProcessor: messageProcessor{
+			marshalizer:      args.Marshalizer,
+			antifloodHandler: args.AntifloodHandler,
+			throttler:        args.Throttler,
+			topic:            args.SenderResolver.RequestTopic(),
+		},
+		baseStorageResolver:  createBaseStorageResolver(args.ValidatorInfoStorage, args.IsFullHistoryNode),
+		validatorInfoPool:    args.ValidatorInfoPool,
+		validatorInfoStorage: args.ValidatorInfoStorage,
+	}, nil
+}
+
+func checkArgs(args ArgValidatorInfoResolver) error {
+	if check.IfNil(args.SenderResolver) {
+		return dataRetriever.ErrNilResolverSender
+	}
+	if check.IfNil(args.Marshalizer) {
+		return dataRetriever.ErrNilMarshalizer
+	}
+	if check.IfNil(args.AntifloodHandler) {
+		return dataRetriever.ErrNilAntifloodHandler
+	}
+	if check.IfNil(args.Throttler) {
+		return dataRetriever.ErrNilThrottler
+	}
+	if check.IfNil(args.ValidatorInfoPool) {
+		return dataRetriever.ErrNilValidatorInfoPool
+	}
+	if check.IfNil(args.ValidatorInfoStorage) {
+		return dataRetriever.ErrNilValidatorInfoStorage
+	}
+
+	return nil
+}
+
+// RequestDataFromHash requests validator info from other peers by hash
+func (res *validatorInfoResolver) RequestDataFromHash(hash []byte, epoch uint32) error {
+	return res.SendOnRequestTopic(
+		&dataRetriever.RequestData{
+			Type:  dataRetriever.HashType,
+			Value: hash,
+			Epoch: epoch,
+		},
+		[][]byte{hash},
+	)
+}
+
+// ProcessReceivedMessage represents the callback func from the p2p.Messenger that is called each time a new message is received
+// (for the topic this validator was registered to, usually a request topic)
+func (res *validatorInfoResolver) ProcessReceivedMessage(message p2p.MessageP2P, fromConnectedPeer core.PeerID) error {
+	err := res.canProcessMessage(message, fromConnectedPeer)
+	if err != nil {
+		return err
+	}
+
+	res.throttler.StartProcessing()
+	defer res.throttler.EndProcessing()
+
+	rd, err := res.parseReceivedMessage(message, fromConnectedPeer)
+	if err != nil {
+		return err
+	}
+
+	switch rd.Type {
+	case dataRetriever.HashType:
+		return res.resolveHashRequest(rd.Value, rd.Epoch, fromConnectedPeer)
```

Comment:
not for this PR: we can implement request by hashes in order to optimize the requests/response mechanism. Will be made just as the transactions resolver

Reply:
somehow I understood from the meeting that we'll only need by hash. Added a TODO for a further PR.